sdk/core/core/src/az_http_policy.c

```diff
@@ -104,5 +105,8 @@ AZ_NODISCARD az_result az_http_pipeline_policy_transport(
   (void)p_policies; // this is the last policy in the pipeline, we just void it
   (void)p_data;
 
+  // make sure the response is resetted
+  _az_http_response_reset(p_response);
```

Comment:
Is this the only place we need to reset the response?

Reply:
Yes